backend/src/server/plugins/serve-ui.ts

```diff
@@ -54,12 +59,35 @@ export const registerServeUI = async (
       schema: {
         hide: true
       },
-      handler: (request, reply) => {
+      handler: async (request, reply) => {
         if (request.url.startsWith("/api")) {
           reply.callNotFound();
           return;
         }
 
+        // all the assets available will be served from the top static serve middleware
+        // if a request is reaching here this only happens due to version skew that happens on deployment
+        if (request.url.startsWith("/assets") && (request.url.endsWith(".js") || request.url.endsWith(".css"))) {
+          const infisicalVersion = await keyStore.getItem(KeyStorePrefixes.InfisicalVersion);
+          const appCfg = getConfig();
+          if (
+            infisicalVersion &&
+            appCfg.INFISICAL_PLATFORM_VERSION &&
+            appCfg.SITE_URL &&
+            infisicalVersion !== appCfg.INFISICAL_PLATFORM_VERSION
+          ) {
+            // this means version skew is happening and container version is old
+            // we wait for 5 second with jitter and then redirect to load balancer again to serve the correct one
+            await delay(applyJitter(5000, 2000));
```

Comment:
if we have 100 containers and 25 of them spin up there is a good chance we keep reaching those 25.  while reduced, doesn't seem to solve the core of the issue. rather feels a work around